src/flow_solver/flow_solver_cases/naca0012.cpp

```diff
@@ -50,15 +51,25 @@ template <int dim, int nstate>
 std::shared_ptr<Triangulation> NACA0012<dim,nstate>::generate_grid() const
 {
     //Dummy triangulation
-    std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation>(
-#if PHILIP_DIM!=1
-            this->mpi_communicator
-#endif
-    );
-    dealii::GridGenerator::Airfoil::AdditionalData airfoil_data;
-    dealii::GridGenerator::Airfoil::create_triangulation(*grid, airfoil_data);
-    grid->refine_global();
-    return grid;
+    if constexpr(dim==2) {
+        std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation>(
+    #if PHILIP_DIM!=1
+                this->mpi_communicator
+    #endif
+        );
+        dealii::GridGenerator::Airfoil::AdditionalData airfoil_data;
+        dealii::GridGenerator::Airfoil::create_triangulation(*grid, airfoil_data);
+        grid->refine_global();
+        return grid;
+    } 
+    else if constexpr(dim==3) {
+        const std::string mesh_filename = this->all_param.flow_solver_param.input_mesh_filename+std::string(".msh");
+        const bool use_mesh_smoothing = false;
+        std::shared_ptr<HighOrderGrid<dim,double>> naca0012_mesh = read_gmsh<dim, dim> (mesh_filename, 0, use_mesh_smoothing);
+        return naca0012_mesh->triangulation;
+    }
+    
+    // TO DO: Avoid reading the mesh twice (here and in set_high_order_grid -- need a default dummy triangulation)
```

Comment:
Can this comment be addressed?

Reply:
I believe that it is the way the code is set up as of now. In "FlowSolver" constructor, we call "create_discontinuous_galerkin" which requires a triangulation coming from the "flow_solver_case", however, when a ".msh" file is used, we must provide a triangulation from the .msh file, hence we need to read the mesh first. Then, when we call "set_higher_order_grid" from "flow_solver_case", we must read it again to return a "HighOrderGrid" for "dg", which we did not return in the first time when we read the mesh since we only returned a triangulation. The solution to this would be to share one single "HighOrderGrid" object for both "FlowSolver" and "dg" which might require some rethinking of how both class is set up to work with "gmsh_reader".

Reply:
Ok, could you open an issue for this and include a link to this discussion?

Reply:
will be resolved by issue https://github.com/dougshidong/PHiLiP/issues/228